### PR TITLE
Add more detailed conda setup instructions to the GATK README

### PR DESCRIPTION
**Pull request by droazen**

### CODE REVIEW

**Comment by droazen** [comment]
@lbergelson / @KevinCLydon , please review for accuracy / completeness of the revised conda setup instructions

File: README.md [hunk 1]
```diff
@@ -81,13 +81,29 @@ releases of the toolkit.
     * GATK4 uses the [Conda](https://conda.io/docs/index.html) package manager to establish and manage the
       Python environment and dependencies required by GATK tools that have a Python dependency. This environment also 
       includes the R dependencies used for plotting in some of the tools. The ```gatk``` environment 
-      requires hardware with AVX support for tools that depend on TensorFlow (e.g. CNNScoreVariant). The GATK Docker image 
+      requires hardware with AVX support for tools that depend on TensorFlow. The GATK Docker image
```
Comment (KevinCLydon) [inline README.md:84]: Are there other tools in GATK that use Tensorflow?

File: README.md [hunk 2]
```diff
@@ -81,13 +81,29 @@ releases of the toolkit.
...
       includes the R dependencies used for plotting in some of the tools. The ```gatk``` environment 
-      requires hardware with AVX support for tools that depend on TensorFlow (e.g. CNNScoreVariant). The GATK Docker image 
+      requires hardware with AVX support for tools that depend on TensorFlow. The GATK Docker image 
       comes with the ```gatk``` environment pre-configured.
-      	* At this time, the only supported platforms are 64-bit Linux distributions. The required Conda environment is not
-	  currently supported on OS X/macOS. 
     * To establish the environment when not using the Docker image, a conda environment must first be "created", and
       then "activated":
-        * First, make sure [Miniconda or Conda](https://conda.io/docs/index.html) is installed (Miniconda is sufficient).
+        * First, make sure [Miniconda or Conda](https://conda.io/docs/index.html) is installed. We recommend installing
```
Comment (KevinCLydon) [inline README.md:88]: Should we add a parenthetical note saying that you might run into issues if you try to use a different conda version?

**Review by KevinCLydon — APPROVED** [review]
I think this is pretty clear.  I have a couple quick questions, but I'm totally happy with this being merged regardless.

File: README.md (hunk 2 above)
Comment (droazen) [reply README.md:88]: Yes, we definitely should!

File: README.md (hunk 1 above)
Comment (droazen) [reply README.md:84]: Yeah, you're right, the mention of Tensorflow will be obsolete once the CNN tools are removed. Pytorch, however, does apparently depend on the Intel MKL, which uses CPU-based features for acceleration:

https://www.reddit.com/r/MachineLearning/comments/iap6yo/discussion_pytorch_favors_intel_against_amds/

File: README.md (hunk 2 above)
Comment (lbergelson) [reply README.md:87]: What is the opt folder and why do we install there?  I've never understood linux folder structures...

File: README.md [hunk 3]
```diff
@@ -79,15 +79,31 @@ releases of the toolkit.
...
+          ```Miniconda3-py310_23.10.0-1``` from [the miniconda download page](https://repo.anaconda.com/miniconda/), selecting the Linux or
+          MacOS version of the installer as appropriate.
+            * This is the same version of ```miniconda``` used by the official GATK docker image.
+            * If you use a different version, you may run into issues.
+            * If you have an ARM-based Mac, you must select the `MacOSX-x86_64` installer, not the `MacOSX-arm64` installer,
+              and rely on Mac OS's built-in x86 emulation.
+        * Set up miniconda:
+            * Install miniconda to a location on your PATH such as ```/opt/miniconda```, and then restart your shell: 
+              ```
```
Comment (lbergelson) [inline README.md:96]: At some point I ended up running `conda init`. I'm not clear on what that did or if it was necessary but maybe it was?

File: README.md (hunk 2 above)
Comment (droazen) [reply README.md:87]: According to one of my AI friends:

The /opt directory in Linux is a reserved space for installing optional or add-on software packages. The name /opt stands for "optional". 

The /opt directory is important for several reasons, including: 

* System management
    * The /opt directory helps keep the Linux system organized and modular, which makes it easier to manage and maintain. 
* Security
    * By separating optional software from the core system, the /opt directory reduces the risk of accidentally modifying or deleting critical system files. 
* Software storage
    * **The /opt directory is often used to store manually compiled software, which is built from source code instead of being installed from distribution repositories.** 
    
Within the /opt directory, applications are usually stored in /opt/bin, while libraries are placed in /opt/lib

File: README.md (hunk 3 above)
Comment (droazen) [reply README.md:96]: hmm.... @KevinCLydon did you have to do that too?

Comment (droazen) [reply README.md:96]: @KevinCLydon doesn't see this command in the history of commands he ran to setup his environment, so I'm going to assume it's not actually necessary.
